Bump Sphinx to 1.8.5 on MSVC

DavidFair opened the pull request:
**Description of work.**
Bumps MSVC to use Sphinx 1.8.5 (matching Ubuntu 18.04). This allows us
to build dev-docs with warnings enabled on both platforms.

RHEL is the odd one out now, but as we have (comparitvely) few
developers on RHEL and this only affects dev docs we can cross that
bridge when Conda is complete

**To test:**
On Windows:
- Build dev-docs-html
- Build docs-html
<!-- Instructions for testing. -->

Pre-requisite of #32222 

Accompanying PR: https://github.com/mantidproject/thirdparty-msvc2015/pull/64

*This does not require release notes* because it's an internal change not visible to users.

<!-- Ensure the base of this PR is correct (e.g. release-next or master)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

DavidFair committed:
Bump Sphinx to 1.8.5 on MSVC

Bumps MSVC to use Sphinx 1.8.5 (matching Ubuntu 18.04). This allows us
to build dev-docs with warnings enabled on both platforms.

RHEL is the odd one out now, but as we have (comparitvely) few
developers on RHEL and this only affects dev docs we can cross that
bridge when Conda is complete

DavidFair committed:
Bump Sphinx to 1.8.5 on MSVC

Bumps MSVC to use Sphinx 1.8.5 (matching Ubuntu 18.04). This allows us
to build dev-docs with warnings enabled on both platforms.

RHEL is the odd one out now, but as we have (comparitvely) few
developers on RHEL and this only affects dev docs we can cross that
bridge when Conda is complete

thomashampson reviewed (commented):
Does this mean we can remove the conditional statements here:
[docs/source/conf.py](https://github.com/mantidproject/mantid/blob/ec346e327616cedf2746fdc860b71ab76e7d9cd4/docs/source/conf.py#L49)
[dev-docs/source/conf.py](https://github.com/mantidproject/mantid/blob/ec346e327616cedf2746fdc860b71ab76e7d9cd4/dev-docs/source/conf.py#L23)

DavidFair committed:
Remove 1.6 compat in dev-docs conf

Removes the compatibility shim in the conf for dev docs. We cannot do
this for our normal docs though, as RHEL-7 still builds them as part of
the nightly build and it uses sphinx <1.8

DavidFair commented:
Good spot, thanks
I've removed the compatibility shim for `dev-docs-html`, but we can't do this for `docs-html` since we still build user docs on RHEL7 as part of packaging. 

Technically #32222  breaks dev doc builds on RHEL but as we have no CI for it, few (if any) developers on it and we intend to bring all platforms onto the same Conda deps we can get away with it.

thomashampson commented on dev-docs/source/conf.py:
```diff
@@ -11,16 +11,10 @@
+def setup(app):
+    """Called automatically by Sphinx when starting the build process
+    """
+    app.add_stylesheet("custom.css")
```
This was renamed in 1.8 (although it wasn't removed until 4.0)
```suggestion
    app.add_css_file("custom.css")
```
https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx.application.Sphinx.add_css_file

DavidFair replied:
Oh I was looking at the man page for 1.6 :man_facepalming:

DavidFair committed:
Use 1.8+ command for Sphinx

Use 1.8+ command for Sphinx instead of 1.6 as incorrectly selected

Co-authored-by: thomashampson <thomas.hampson@stfc.ac.uk>

martyngigg commented:
I've just realised I've caused a problem here. I never merged mantidproject/thirdparty-msvc2015#63 so your branch in `thirdparty` is not based off a commit that has the changes form the above PR. If I merge the above PR can you rebase your third-party branch and update the SHA here?

martyngigg reviewed (changes requested):
I'm going to mark this requesting changes to avoid it being merged by accident.

DavidFair commented:
Thanks for the heads up, this PR and the one in the third party repo should now be updated

martyngigg commented:
Thanks. I'll give it a quick retest and merge if okay
